src/components/Popup/Popup.js

```diff
@@ -46,64 +50,92 @@ const propTypes = {
   className: PropTypes.string,
 
   /**
-   * Function triggered on close button click.
-   */
-  onCloseClick: PropTypes.func,
-
-  /**
-   * Function triggered on key up.
+   * Title HTML attributes.
    */
-  onKeyUp: PropTypes.func,
+  titles: PropTypes.shape({
+    closeButton: PropTypes.string,
+  }),
 
   /**
-   * Popup title.
+   * Function triggered on close button click.
    */
-  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  onCloseClick: PropTypes.func,
 
   /**
-   * Popup padding.
+   * Hide or show the header.
    */
-  padding: PropTypes.string,
+  showHeader: PropTypes.bool,
 
   /**
-   * HTML tabIndex attribute.
+   * Hide or show close button.
    */
-  tabIndex: PropTypes.string,
+  showCloseButton: PropTypes.bool,
 
   /**
-   * Hide or show the header.
+   * Render the header
    */
-  showHeader: PropTypes.bool,
+  renderHeader: PropTypes.func,
 
   /**
-   * Hide or show close button.
+   * Render the close button
    */
-  showCloseButton: PropTypes.bool,
+  renderCloseButton: PropTypes.func,
 
   /**
-   * Translation function.
-   * @param {function} Translation function returning the translated string.
+   * Render the footer
    */
-  t: PropTypes.func,
+  renderFooter: PropTypes.func,
 };
 
 const defaultProps = {
+  title: null,
   feature: null,
   panIntoView: false,
   panRect: null,
   popupCoordinate: null,
   className: 'rs-popup',
   showCloseButton: true,
   showHeader: true,
-  title: null,
-  padding: '10px',
-  tabIndex: '',
-  onKeyUp: () => {},
+  titles: { closeButton: 'Close' },
   onCloseClick: () => {},
-  t: p => p,
+  renderHeader: null,
+  renderCloseButton: null,
+  renderFooter: () => null,
 };
 
 class Popup extends PureComponent {
```

Comment:
Why not a functional component?

Reply:
I've tried but there was a problem with the refresh and the panIntoView option. @mhaertwig fixed it in the version so I choose to let it like that since I couldn't succeed to have a better version.